core/src/main/java/com/linecorp/armeria/common/AbstractRequestContextBuilder.java

```diff
@@ -149,6 +139,27 @@ protected AbstractRequestContextBuilder(boolean server, RpcRequest rpcReq, URI u
         query = pathAndQuery.query();
     }
 
+    private static SessionProtocol getSessionProtocol(URI uri) {
+        final String schemeStr = uri.getScheme();
+        if (schemeStr != null && schemeStr.indexOf('+') < 0) {
+            final SessionProtocol parsed = SessionProtocol.find(schemeStr);
+            if (parsed == null) {
+                return reportInvalidScheme(uri);
```

Comment:
I think I saw this pattern in another PR and found it a bit weird, and now am reconsidering it again. Is it clearer to use `throw newInvalidSchemeException(uri)` so it's easy to see that this line is throwing an exception, without much change in verbosity (actually guess less characters)?

Reply:
Fixed :-)